packages/mds-compliance-service/repository/index.ts

```diff
@@ -146,42 +146,61 @@ class ComplianceReadWriteRepository extends ReadWriteRepository {
     }
   }
 
+  // TODO add a beforeCommit fn param
   public createComplianceSnapshot = async (
-    complianceSnapshot: ComplianceSnapshotDomainModel
+    complianceSnapshot: ComplianceSnapshotDomainModel,
+    beforeCommit: (domain: ComplianceSnapshotDomainModel) => Promise<void>
```

Comment:
I think when using this pattern in other places, the `beforeCommit` was optional. I suppose we could pass `() => {}` or something, but might be a good idea to make it optional and then just:
```ts
if (beforeCommit) {
   await beforeCommit(domain)
}
```